cli/ballerina-cli/src/main/java/io/ballerina/cli/task/RunBuildToolsTask.java

```diff
@@ -207,6 +211,17 @@ public void execute(Project project) {
                 throw createLauncherException(e.getMessage());
             }
         }
+
+        // Exit if there is any error diagnostic
+        boolean hasErrors = false;
+        for (Diagnostic d : toolDiagnostics) {
+            if (d.diagnosticInfo().severity().equals(DiagnosticSeverity.ERROR)) {
+                hasErrors = true;
+            }
+        }
```

Comment:
We can simply this using functional style expression:
```suggestion
boolean hasErrors = toolDiagnostics.stream()
    .anyMatch(d -> d.diagnosticInfo().severity().equals(DiagnosticSeverity.ERROR));
```